lib/providers/config_provider.dart

```diff
@@ -1,142 +1,370 @@
-import 'package:cogboardmobileapp/models/widgets_model.dart';
+import 'package:cogboardmobileapp/models/widget_status_change_model.dart';
 import 'package:cogboardmobileapp/utils/shared_preferences_utils.dart';
+import 'package:enum_to_string/enum_to_string.dart';
 import 'package:flutter/material.dart';
 import 'package:http/http.dart' as http;
 
 class ConfigProvider with ChangeNotifier {
   Config _config;
   List<Board> _boards;
-  UrlPreferences _urlPreferences;
-  String _currentUrl = 'http://150.254.30.119/api/config';
+  SettingsPreferences _settingsPreferences;
   int _snackBarsToRemove = 0;
+  bool _webSocketConnectionErrorPresent = false;
+  List<DashboardWidget> _lastNotificationUpdateWidgetsState;
+  String _notificationPayload;
+  List<WidgetStatusChange> _widgetsInNotificationPayload = [];
+  Board _currentBoard;
+  http.Client client = http.Client();
 
   ConfigProvider() {
-    _urlPreferences = new UrlPreferences(favouriteWidgetIds: [], quarantineWidgetIds: []);
-    checkIfQuarantineExpirationDateHasExceeded();
-    new Timer.periodic(const Duration(hours: 1), everyHourCheckTimer);
+    new Timer.periodic(const Duration(minutes: 1), everyMinuteCheckTimer);
   }
 
-  void everyHourCheckTimer(Timer timer) async {
-    await checkIfQuarantineExpirationDateHasExceeded();
+  ConfigProvider withSettingsPreferences(SettingsPreferences settingsPreferences) {
+    _settingsPreferences = settingsPreferences;
+    notifyListeners();
+    return this;
+  }
+
+  void everyMinuteCheckTimer(Timer timer) async {
+    if (currentConnection != null) {
+      await checkIfQuarantineExpirationDateHasExceeded();
+    }
   }
 
   Future<void> checkIfQuarantineExpirationDateHasExceeded() async {
     DateTime currentTime = new DateTime.now();
-    print(currentTime);
-    if(await SharedPref.containsKey(Widgets.QUARANTINE_WIDGETS_EXPIRATION_DATE_KEY)) {
-      DateTime expirationDate = DateTime.parse(await SharedPref.read((Widgets.QUARANTINE_WIDGETS_EXPIRATION_DATE_KEY)));
-      if(currentTime.day > expirationDate.day) {
-        await SharedPref.save(Widgets.QUARANTINE_WIDGETS_EXPIRATION_DATE_KEY, currentTime.toString());
-        await(removeQuarantineWidgets());
+    bool someWidgetsWereRemoved = false;
+    currentConnection.quarantineWidgets.removeWhere((widget) {
+      bool shouldRemoveWidget = widget.expirationDate != null ? widget.expirationDate.day < currentTime.day : false;
+      if (shouldRemoveWidget) {
+        someWidgetsWereRemoved = true;
       }
-    } else {
-      await SharedPref.save(Widgets.QUARANTINE_WIDGETS_EXPIRATION_DATE_KEY, currentTime.toString());
+      return shouldRemoveWidget;
+    });
+    if (someWidgetsWereRemoved) {
+      await SharedPref.save(SettingsPreferences.KEY, jsonEncode(_settingsPreferences.toJson()));
+      notifyListeners();
     }
   }
 
+  Future<void> fetchConfig() async {
+      final response = await client.get('http://$currentUrl/api/config');
```

Comment:
🥇 [NOW] `http://$currentUrl/api/config` what if app runs on `https`? 

I think that by default we should use https (if url is provided without http/https)

pseudo code:
`currentUrl.startsWith('http') ? '$currentUrl/api/config' : 'https://$currentUrl/api/config'`

Reply:
this should be done for all endpoint calls - currentUrl here already has this logic applied

Reply:
done

Reply:
done